src/main/java/mmosii/bookstore/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package mmosii.bookstore.service.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mmosii.bookstore.dto.category.CategoryDto;
+import mmosii.bookstore.dto.category.CreateCategoryRequestDto;
+import mmosii.bookstore.exception.EntityNotFoundException;
+import mmosii.bookstore.mapper.CategoryMapper;
+import mmosii.bookstore.model.Category;
+import mmosii.bookstore.repository.book.CategoryRepository;
+import mmosii.bookstore.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll()
+                .stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        return categoryRepository
+                .findById(id)
+                .map(categoryMapper::toDto)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find category by id " + id));
+    }
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto categoryDto) {
+        return categoryMapper.toDto(categoryRepository.save(categoryMapper.toEntity(categoryDto)));
```

Comment:
I would define a separate variable for `toEntity` - this would improve readability